src/Confluent.Kafka/Internal/Util.cs

```diff
@@ -28,19 +29,31 @@ internal static class Marshal
             /// <summary>
             ///     Interpret a zero terminated c string as UTF-8.
             /// </summary>
-            public static string PtrToStringUTF8(IntPtr strPtr)
+            public unsafe static string PtrToStringUTF8(IntPtr strPtr)
             {
-                // TODO: Is there a built in / vectorized / better way to implement this?
-                var length = 0;
-                unsafe
-                {
-                    byte* pTraverse = (byte*)strPtr;
-                    while (*pTraverse != 0) { pTraverse += 1; }
-                    length = (int)(pTraverse - (byte*)strPtr);
-                }
+                // TODO: Is there a built in / vectorized / better way to implement this?              
+                byte* pTraverse = (byte*)strPtr;
+                while (*pTraverse != 0) { pTraverse += 1; }
+                var length = (int)(pTraverse - (byte*)strPtr);
+#if NET45
                 var strBuffer = new byte[length];
                 System.Runtime.InteropServices.Marshal.Copy(strPtr, strBuffer, 0, length);
                 return Encoding.UTF8.GetString(strBuffer);
+#else
+                // avoid unnecessary data copying on NET45+
+                return Encoding.UTF8.GetString((byte*)strPtr.ToPointer(), length);
+#endif
+            }
+
+            /// <summary>
```

Comment:
can you change the indentation / new lines to match convention of other comments, then I think this PR is good to merge - thanks!

Reply:
done! kindly review